Userland/Libraries/LibVideo/Track.h

```diff
@@ -11,6 +11,7 @@
 #include <AK/Traits.h>
 #include <AK/Types.h>
 #include <AK/Variant.h>
+#include <AK/kmalloc.h>
```

Comment:
Cursory glance - I don't see any symbol from kmalloc.h being used here?

Reply:
Variant.h uses placement new in some methods, which are not always generated. Importing it in Variant.h causes lots of new inclusions, but here it's actually triggered.

Reply:
Yeah I disagree with this approach then. If Variant.h needs a header, it should include it itself, it doesn't make sense for a random LibVideo header to have to resolve Variant's non-forward-declarable dependencies for it.

What header does Variant.h *actually* need? Because it doesn't use any kmalloc.h symbols either. If the error was due to placement new, I'd assume `<new>`, for which we have existing examples on how to keep `<new>` out of the kernel (e.g. Singleton.h).

Reply:
Putting it into Variant.h is fine with me.

Trying to include `<new>` is indeed more direct, and causes the exact problems that you suggested. However, implementing any `#ifdef`-logic ad-hoc in Variant.h feels like duplicating the `#ifdef`s in kmalloc.h, which seems to be a bad idea.

Would including kmalloc.h in Variant.h work for you? This seems to be the closest practical solution to what you suggested.

Reply:
Actually, it's less complicated logic than I expected:
```c++
#if defined(KERNEL)
#    include <Kernel/Heap/kmalloc.h>
#else
#    include <new>
#endif
```

It still seems weird to me, I'd prefer just putting `<kmalloc.h>` in Variant.h